api/src/main/java/io/minio/MinioClient.java

```diff
@@ -815,8 +819,8 @@ public boolean verify(String hostname, SSLSession session) {
    * @param length         Length of HTTP request body.
    */
   private Request createRequest(Method method, String bucketName, String objectName,
-                                String region, Map<String,String> headerMap,
-                                Map<String,String> queryParamMap, final String contentType,
+                                String region, Map<String,String[]> headerMap,
+                                Map<String,String[]> queryParamMap, final String contentType,
```

Comment:
Can we use `Multimap` here than `Map<String,String[]>` ?
```java
import com.google.common.collect.TreeMultimap;
import com.google.common.collect.Multimap;

public class Hello {
  public static void main(String[] args) throws Exception {
    Multimap<String, String> map = TreeMultimap.create();
    map.put("key1", "value3");
    map.put("key1", "value1");
    map.put("key1", "value2");
    // prints [value1, value2, value3]
    System.out.println(map.get("key1"));
  }
}
```

Reply:
Fixed, but I decided to use HashMultimap instead of TreeMultimap (since we don't need to order keys and values)